test/ShadowingTest.hs

```diff
@@ -14,11 +14,11 @@ tests =
     , testCase "Abs with underscore" $
         checkShadowing ["_x"] abs_with_underscore @?= []
     , testCase "Abs twice shadow" $
-        checkShadowing [] abs_twice_shadow @?= ["y", "x"]
+        checkShadowing [] abs_twice_shadow @?= [(ShadowVar "y" lit), (ShadowVar "x" lit)]
```

Comment:
@DaQuirm, here I need only `Identifier` from `ShadowVar`. How can I beautifully unpack it here & compare?

Reply:
You could write a function: `ShadowVar -> identifier` and `map` it. Or you could make `ShadowVar` a record and get this function for free.